client/state/site-rename/actions.js

```diff
@@ -54,9 +62,27 @@ export const requestSiteRename = ( siteId, newBlogName, discard ) => dispatch =>
 			} );
 		} )
 		.catch( error => {
+			if ( error.statusCode === 500 ) {
+				dispatch( getErrorNotice( translate( 'Sorry, something generic (500)' ) ) );
```

Comment:
Note: The copy of each of these messages will need review - I've purposely left their corresponding codes in the messages so that they get addressed before merge :)